src/Extension/DateTimeExtension.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Faker\Core\Extension;
+
+/**
+ * FakerPHP extension for Date-related randomization.
+ *
+ * Functions accepting a date string use the `strtotime()` function internally.
+ *
+ * @experimental
+ *
+ * @since 1.20.0
+ */
+interface DateTimeExtension extends Extension
+{
+    /**
+     * Get a DateTime object between January 1, 1970, and `$until` (defaults to "now").
+     *
+     * @param \DateTime|int|string $until    maximum timestamp, defaults to "now"
+     * @param string|null          $timezone zone timezone for generated date, fallback to `DateTime::$defaultTimezone` and `date_default_timezone_get()`.
+     *
+     * @see \DateTimeZone
+     * @see http://php.net/manual/en/timezones.php
+     * @see http://php.net/manual/en/function.date-default-timezone-get.php
+     *
+     * @example DateTime('2005-08-16 20:39:21')
+     */
+    public function dateTime($until = 'now', string $timezone = null): \DateTime;
+
+    /**
+     * Get a DateTime object for a date between January 1, 0001, and now.
+     *
+     * @param \DateTime|int|string $until    maximum timestamp, defaults to "now"
+     * @param string|null          $timezone zone timezone for generated date, fallback to `DateTime::$defaultTimezone` and `date_default_timezone_get()`.
+     *
+     * @example DateTime('1265-03-22 21:15:52')
+     *
+     * @see http://php.net/manual/en/timezones.php
+     * @see http://php.net/manual/en/function.date-default-timezone-get.php
+     */
+    public function dateTimeAD($until = 'now', string $timezone = null): \DateTime;
+
+    /**
+     * Get a DateTime object a random date between `$from` and `$until`.
+     * Accepts date strings that can be recognized by `strtotime()`.
+     *
+     * @param \DateTime|string     $from     defaults to 30 years ago
+     * @param \DateTime|int|string $until    maximum timestamp, defaults to "now"
+     * @param string|null          $timezone zone timezone for generated date, fallback to `DateTime::$defaultTimezone` and `date_default_timezone_get()`.
+     *
+     * @see \DateTimeZone
+     * @see http://php.net/manual/en/timezones.php
+     * @see http://php.net/manual/en/function.date-default-timezone-get.php
+     */
+    public function dateTimeBetween($from = '-30 years', $until = 'now', string $timezone = null): \DateTime;
```

Comment:
```suggestion
    public function dateTimeBetween(string $from = '-30 years', string $until = 'now', string $timezone = null): \DateTime;
```
I do wonder though if it isnt better to just add a DateTimeInterface in the from and untill